utility/importer/brave_importer.cc

```diff
@@ -315,3 +338,233 @@ void BraveImporter::ImportStats() {
 
   bridge_->UpdateStats(stats);
 }
+
+bool ParseWalletPassphrase(BraveLedger& ledger,
+  const base::Value& session_store_json) {
+  const base::Value* wallet_passphrase_value =
+    session_store_json.FindPathOfType(
+      {"ledger", "info", "passphrase"},
+      base::Value::Type::STRING
+    );
+  if (!wallet_passphrase_value) {
+    LOG(ERROR) << "Wallet passphrase not found in session-store-1";
+    return false;
+  }
+
+  ledger.passphrase = wallet_passphrase_value->GetString();
+  return !ledger.passphrase.empty();
+}
+
+bool TryFindBoolKey(const base::Value* dict,
+  const std::string key, bool& value_to_set) {
+  auto* value_read = dict->FindKeyOfType(key, base::Value::Type::BOOLEAN);
+  if (value_read) {
+    value_to_set = value_read->GetBool();
+    return true;
+  }
+  return false;
+}
+
+bool TryFindStringKey(const base::Value* dict,
+  const std::string key, std::string& value_to_set) {
+  auto* value_read = dict->FindKeyOfType(key, base::Value::Type::STRING);
+  if (value_read) {
+    value_to_set = value_read->GetString();
+    return true;
+  }
+  return false;
+}
+
+bool TryFindIntKey(const base::Value* dict, const std::string key,
+  int& value_to_set) {
+  auto* value_read = dict->FindKeyOfType(key, base::Value::Type::INTEGER);
+  if (value_read) {
+    value_to_set = value_read->GetInt();
+    return true;
+  }
+  return false;
+}
+
+bool ParsePaymentsPreferences(BraveLedger& ledger,
+  const base::Value& session_store_json) {
+  const base::Value* settings = session_store_json.FindKeyOfType(
+    "settings",
+    base::Value::Type::DICTIONARY);
+  if (!settings) {
+    LOG(ERROR) << "No entry \"settings\" found in session-store-1";
+    return false;
+  }
+
+  auto* payments = &ledger.settings.payments;
+
+  TryFindBoolKey(settings, "payments.enabled", payments->enabled);
+  TryFindBoolKey(settings, "payments.allow-non-verified-publishers",
+    payments->allow_non_verified);
+  TryFindBoolKey(settings, "payments.allow-media-publishers",
```

Comment:
I think it would be preferable to explicitly validate the imported state here (and throughout). Since you sometimes ignore the return values of the `TryFind*` functions, what happens when the state you are importing from is invalid? I haven't confirmed this, but it seems that in some cases you might end up with indeterminate values for some of the fields of the BraveLedger struct and substructs.